cmake/sysbuild/fast_pair_hex.cmake

```diff
@@ -47,6 +47,26 @@ function(fast_pair_hex_pm)
     )
 endfunction()
 
+# This function calculates the absolute partition address by summing up
+# all hex numbers preceded by @ in the `partition_nodelabel` string.
+# The result is stored in the variable provided as `absolute_addr`.
+# This function should probably one day be moved to some common utilities file.
+function(get_partition_absolute_addr absolute_addr partition_nodelabel)
+  set(addr_regex_pattern "@([0-9a-fA-F]+)")
+  # Use regex to match all hex numbers preceded by @
+  string(REGEX MATCHALL ${addr_regex_pattern} matches ${partition_nodelabel})
+
+  set(result 0x0)
+
+  # Extract the addresses from the matches and add them
```

Comment:
It won't work properly for non-partition nodes that support more complex mapping between addresses. In that case the `ranges` property must be used.

Reply:
For example it won't work for this Atmel node, because it uses different address mapping:
```
pinctrl: pinctrl@400e0e00 {
			compatible = "atmel,sam-pinctrl";
			#address-cells = <1>;
			#size-cells = <1>;
			ranges = <0x400e0e00 0x400e0e00 0xc00>;

			pioa: pio@400e0e00 {
				compatible = "atmel,sam-gpio";
				reg = <0x400e0e00 0x190>;
				interrupts = <11 1>;
				clocks = <&pmc PMC_TYPE_PERIPHERAL 11>;
				gpio-controller;
				#gpio-cells = <2>;
				#atmel,pin-cells = <2>;
			};
```

Reply:
Thanks. In that case, I would leave as it is